Sources/PackageModel/ResolvedModels.swift

```diff
@@ -24,9 +24,13 @@ public final class ResolvedModule: CustomStringConvertible, ObjectIdentifierProt
     /// The direct dependencies of this module.
     public let dependencies: [ResolvedModule]
 
+    /// The product dependencies of this module.
+    public let productDependencies: [ResolvedProduct]
+
     /// The transitive closure of the module dependencies, in build order.
     public lazy var recursiveDependencies: [ResolvedModule] = {
-        return try! topologicalSort(self.dependencies, successors: { $0.dependencies })
+        let moduleDependencies = self.dependencies + self.productDependencies.flatMap{$0.modules}
+        return try! topologicalSort(moduleDependencies, successors: { $0.recursiveDependencies })
```

Comment:
This is a bit weird to run `.recursiveDependencies` within the successor function -- the expected use would be that each call just returns the immediate successors.